lib/src/main/java/com/basistheory/elements/service/BasisTheoryElements.kt

```diff
@@ -90,6 +96,68 @@ class BasisTheoryElements internal constructor(
             throw ApiException(e.code, e.responseHeaders, e.responseBody, e.message)
         }
 
+    fun encryptTokens(
+        encryptTokenRequest: EncryptTokenRequest,
+    ): Array<EncryptTokenResponse> =
+        try {
+            val processedTokenRequests = processTokenRequests(encryptTokenRequest.tokenRequests)
+
+            processedTokenRequests.map { tokenData ->
+                val encrypted = JWEEncryption.encrypt(
+                    tokenData.first,
+                    encryptTokenRequest.publicKey,
+                    encryptTokenRequest.keyId
+                )
+                EncryptTokenResponse(encrypted, tokenData.second)
+            }.toTypedArray()
+        } catch (e: IllegalArgumentException) {
+            throw e
+        } catch (e: Exception) {
+            throw EncryptTokenException("Failed to encrypt tokens", e)
+        }
+
+    private fun processTokenRequests(tokenRequests: Any): List<Pair<Any, String>> {
+        val tokenRequestsMap = tokenRequests.toMap()
+
+        val isSingleTokenRequest = tokenRequestsMap.containsKey("type")
+        if (isSingleTokenRequest) {
+            return listOf(processIndividualTokenRequest(tokenRequestsMap))
+        }
+
+        return processMultipleTokenRequests(tokenRequestsMap)
+    }
+
+    private fun processMultipleTokenRequests(tokenRequestsMap: Map<String, Any?>): List<Pair<Any, String>> {
+        return tokenRequestsMap.values.map { tokenRequest ->
+            processIndividualTokenRequest(tokenRequest?.toMap())
+        }
+    }
+
+    private fun processIndividualTokenRequest(tokenRequest: Any?): Pair<Any, String> {
+        val rawData = tokenRequest?.tryGetValue<Any>("data")
+        val rawType = tokenRequest?.tryGetValue<String>("type")
+
+        validateTokenRequestFields(rawData, rawType)
+
+        return Pair(processTokenData(rawData)!!, rawType!!)
+    }
+
+    private fun validateTokenRequestFields(data: Any?, type: String?) {
+        if (data == null || type == null) {
+            throw IllegalArgumentException("Both token data and type must be provided")
+        }
+    }
+
+    private fun processTokenData(data: Any?): Any? {
+        return if (data == null) null
+        else if (data::class.java.isPrimitiveType()) data
+        else if (data is TextElement) data.tryGetTextToTokenize()
+            .toValueType(data.getValueType)
+        else if (data is ElementValueReference) data.getValue()
+            .toValueType(data.getValueType)
+        else replaceElementRefs(data.toMap())
```

Comment:
You can use `when` instead of `if/else` to improve readability

```kotlin
private fun processTokenData(data: Any?): Any? = when (data) {
    null -> null
    is TextElement -> data.tryGetTextToTokenize().toValueType(data.getValueType)
    is ElementValueReference -> data.getValue().toValueType(data.getValueType)
    else -> if (data::class.java.isPrimitiveType()) data else replaceElementRefs(data.toMap())
}
```